[vtk] Fix type error in ternary operator

This didn't show up until Drake started using -std=c++23.

--- Common/ExecutionModel/vtkAlgorithm.cxx
+++ Common/ExecutionModel/vtkAlgorithm.cxx
@@ -1121,7 +1121,7 @@ void vtkAlgorithm::SetInputConnection(int port, vtkAlgorithmOutput* input)
   // The connection is not present.
   vtkDebugMacro("Setting connection to input port index "
     << consumerPort << " from output port index " << producerPort << " on algorithm "
-    << (producer ? producer->GetObjectDescription() : nullptr) << ".");
+    << (producer ? producer->GetObjectDescription() : std::string("nullptr")) << ".");
 
   // Add this consumer to the new input's list of consumers.
   if (newInfo)
